java/arcs/android/devTools/DevToolsService.kt

```diff
@@ -0,0 +1,71 @@
+package arcs.android.devTools
+
+import android.app.Service
+import android.content.Intent
+import android.os.Binder
+import android.os.Bundle
+import android.os.Handler
+import android.os.HandlerThread
+import android.os.IBinder
+import android.os.Looper
+import android.os.Message
+
+class DevToolsService : Service() {
+
+    // Binder given to clients
+    protected val binder = LocalBinder()
+    protected var serviceLooper: Looper? = null
+    protected var serviceHandler: ServiceHandler? = null
+    protected var webSocket: DevWebSocket? = null
+
+    /**
+     * Class used for the client Binder.
+     */
+    inner class LocalBinder : Binder() {
+        // Return this instance of LocalService so clients can call public methods
+        fun getService(): DevToolsService = this@DevToolsService
+    }
+
+    fun send(str: String) {
+        val bun = Bundle()
+        bun.putString("message", str)
+        val msg = Message()
+        msg.setData(bun)
+        serviceHandler?.sendMessage(msg)
+    }
+
+    override fun onBind(intent: Intent): IBinder? {
+        webSocket?.start()
+        return binder
+    }
+
+    override fun onStartCommand(intent: Intent, flags: Int, startId: Int): Int {
+        serviceHandler?.obtainMessage()?.also { msg ->
+            msg.arg1 = startId
+            serviceHandler?.sendMessage(msg)
+        }
+
+        return START_STICKY
+    }
+
+    protected inner class ServiceHandler(looper: Looper) : Handler(looper) {
+
+        override fun handleMessage(msg: Message) {
+            try {
+                webSocket?.send(msg.data.getString("message", ""))
+            } catch (e: InterruptedException) {
+                // Restore interrupt status.
+                Thread.currentThread().interrupt()
+            }
+        }
+    }
+
+    override fun onCreate() {
+        HandlerThread("ServiceStartArguments").apply {
+            start()
+            serviceLooper = looper
```

Comment:
All these fields being nullable is dirty... maybe Jason will have some advice?

Reply:
I moved a bunch of the variables to `lateinit`